### PR TITLE
GODRIVER-3278 Port OIDC machine callback to master

### PR DESCRIPTION
**Pull request by blink1073**
GODRIVER-3278

## Summary

Port OIDC functionality to master.

### CODE REVIEW

File: mongo/collection.go [hunk 1]
```diff
@@ -1304,6 +1305,7 @@ func (coll *Collection) find(ctx context.Context, filter interface{},
 
 	f, err := marshal(filter, coll.bsonOpts, coll.registry)
 	if err != nil {
+		fmt.Println(err)
```
Comment (matthewdale) [inline mongo/collection.go:1308]: Remove debugging `Println`.
```suggestion
```

P.S. Looks like this is in `v1` also. I'll open a PR to remove that line in `v1`.

**Review by matthewdale — CHANGES_REQUESTED** [review]

File: mongo/collection.go (hunk 1 above)
Comment (matthewdale) [reply mongo/collection.go:1308]: PR for v1: https://github.com/mongodb/mongo-go-driver/pull/1722

**Review by matthewdale — DISMISSED** [review]
Looks good! 👍

**Comment by blink1073** [comment]
I had to address a few import and lint failures since your review.

**Review by matthewdale — APPROVED** [review]

**Pull request merged**